### PR TITLE
Updated MSAL-Node from v1.x -> v2.x

### PR DESCRIPTION
**Pull request by Robbie-Microsoft**
Updated MSAL-Node from v1.x -> v2.x and fixed bugs that arose as a result of the update.

Fixed unrelated bug in both `4-AccessControl` samples where the TODO list's `delete` wasn't functoning correctly.

### CODE REVIEW

File: 4-AccessControl/1-app-roles/App/package-lock.json [hunk 1]
```diff
@@ -25,11 +25,11 @@
         "supertest": "^6.1.4"
       }
     },
-    "../../../common/msal-node-wrapper": {
```
Comment (bgavrilMS) [inline 4-AccessControl/1-app-roles/App/package-lock.json:28]: How about you delete the package-lock.json? This way the sample will always use the latest version of MSAL.

**Review by bgavrilMS — APPROVED** [review]

File: 4-AccessControl/1-app-roles/App/package-lock.json (hunk 1 above)
Comment (Robbie-Microsoft) [reply 4-AccessControl/1-app-roles/App/package-lock.json:28]: Oh, shoot! I misread the GitHub Issue connected to this PR. I'll delete them.

Comment (Robbie-Microsoft) [reply 4-AccessControl/1-app-roles/App/package-lock.json:28]: The build scripts fail when there is no package-lock.json

File: 4-AccessControl/2-security-groups/App/routes/todolistRoutes.js [hunk 2]
```diff
@@ -5,8 +5,7 @@
 const router = express.Router();
 
 // user routes
-router.get('/', todolistController.getTodos);
+router.get('', todolistController.getTodos);
```
[the remark this answers is not in the document]
Comment (Robbie-Microsoft) [reply 4-AccessControl/2-security-groups/App/routes/todolistRoutes.js]: fixed

File: 4-AccessControl/1-app-roles/App/package-lock.json (hunk 1 above)
Comment (bgavrilMS) [reply 4-AccessControl/1-app-roles/App/package-lock.json:28]: Can you fix them?

Comment (Robbie-Microsoft) [reply 4-AccessControl/1-app-roles/App/package-lock.json:28]: Per offline discussion, this will be a separate task.

**Pull request merged**